stdlib/public/Concurrency/Task.cpp

```diff
@@ -174,23 +174,31 @@ static void destroyJob(SWIFT_CONTEXT HeapObject *obj) {
   assert(false && "A non-task job should never be destroyed as heap metadata.");
 }
 
-SWIFT_CC(swift)
-static void destroyTask(SWIFT_CONTEXT HeapObject *obj) {
-  auto task = static_cast<AsyncTask*>(obj);
+AsyncTask::~AsyncTask() {
   // For a future, destroy the result.
-  if (task->isFuture()) {
-    task->futureFragment()->destroy();
+  if (isFuture()) {
+    futureFragment()->destroy();
   }
 
   // Release any objects potentially held as task local values.
-  task->Local.destroy(task);
+  Local.destroy(this);
+}
+
+SWIFT_CC(swift)
+static void destroyTask(SWIFT_CONTEXT HeapObject *obj) {
+  auto task = static_cast<AsyncTask*>(obj);
+
+  task->~AsyncTask();
```

Comment:
no idea if doing a destructor is nicer... am I missing something? 
so far we've mostly had all destroyX things explicitly (but I'm by no means C++ or this codebase guru...)

Reply:
I think it's a matter of taste